types/src/contract_event.rs

```diff
@@ -262,6 +338,37 @@ impl ContractEventV2 {
     }
 }
 
+impl Serialize for ContractEventV2 {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        const NUM_FIELDS: usize = 2;
+        let mut s = serializer.serialize_struct("ContractEventV2", NUM_FIELDS)?;
+        s.serialize_field("type_tag", &self.type_tag)?;
+        s.serialize_field("event_data", &serde_bytes::Bytes::new(&self.event_data))?;
+        s.end()
```

Comment:
I mean.. bring the placeholder type out, and construct it here and serialize the placeholder.. (maybe call it EventV2Inner, EventV2Persisted, etc)

(Is it really worth it to cache the size? 🤦‍♂️ or maybe simply make it lazy by using a `#[serde(skip)]`ed `OnceCell`?)

----
like this https://play.rust-lang.org/?version=stable&mode=debug&edition=2024&gist=7e7e007b76dc3c4fecbccf916c6990d8

Reply:
My main goal was to avoid unwrap for `size()` for gas charging at execution time, are you concerned about size reconstruction on deserialization? We can just make it `Option<usize>` and let it be `None` when deserialized?

Reply:
Otherwise no need to cache size (it's just we will be doing type tag serialization twice -> at construction of event and at gas charging point)

Reply:
I see.. 
> are you concerned about size reconstruction on deserialization?

no, just pull the Placeholder struct out and use it for both ser and de

> Otherwise no need to cache size (it's just we will be doing type tag serialization twice -> at construction of event and at gas charging point)

it's okay too, it called serialized_size, not really serializing